server/src/main/java/org/elasticsearch/discovery/PeerFinder.java

```diff
@@ -152,8 +152,11 @@ private boolean assertInactiveWithNoKnownPeers() {
     }
 
     PeersResponse handlePeersRequest(PeersRequest peersRequest) {
+        if (peersRequest.getSourceNode().equals(getLocalNode())) {
```

Comment:
why can't this be an assertion anymore?

Reply:
Local nodes with multiple addresses were getting here, but I thought of a better approach in 0958cdb2964.